mods/farming/api.lua

```diff
@@ -252,32 +216,19 @@ farming.grow_plant = function(pos, elapsed)
 end
 
 -- Register plants
-farming.register_plant = function(name, def)
-	local mname = name:split(":")[1]
-	local pname = name:split(":")[2]
+function farming.register_plant(name, def)
+	local mname, pname = unpack(name:split(":"))
 
 	-- Check def table
-	if not def.description then
-		def.description = S("Seed")
-	end
 	if not def.harvest_description then
 		def.harvest_description = pname:gsub("^%l", string.upper)
 	end
-	if not def.inventory_image then
-		def.inventory_image = "unknown_item.png"
-	end
-	if not def.steps then
-		return nil
-	end
-	if not def.minlight then
-		def.minlight = 1
-	end
-	if not def.maxlight then
-		def.maxlight = 14
-	end
-	if not def.fertility then
-		def.fertility = {}
-	end
+	assert(def.description, "Missing description for " .. name)
+	assert(def.inventory_image, "Missing inventory_image for " .. name)
+	assert(def.steps, "Missing number of steps for " .. name)
+	def.minlight = def.minlight or 1
+	def.maxlight = def.maxlight or 14
+	def.fertility = def.fertility or {}
```

Comment:
With `{}` as fertility, the seed will never grow.
`{"grassland"}` would be a reasonable default.
`fertility` isn't even documented.
I suggest doing an assert and documenting.

(Btw. at some places pairs is used on that table, `ipairs` would make more sense, see `init.lua` for example values.)

Reply:
fixed